### PR TITLE
13854 crisis line fix

### PR DESCRIPTION
**Pull request by rianfowler**
## Description
fix the crisis line modal on team sites 

## Testing done
verified locally by proxying new assets 

## Screenshots team sites 
screen shot 2018-10-04 at 17 39 35

## screenshots va.gov (original)
screen shot 2018-10-04 at 17 39 39

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/proxy-rewrite/proxy-rewrite-entry.jsx [hunk 1]
```diff
@@ -98,5 +100,11 @@ document.addEventListener('DOMContentLoaded', _e => {
   // startLRNHealthCarWidget(commonStore);
   startFeedbackWidget(commonStore);
   // startAnnouncementWidget(commonStore);
-  startVAFooter(commonStore);
+  startVAFooter({
+    handleFooterDidMount: () => {
+      addOverlayTriggers();
+      addFocusBehaviorToCrisisLineModal();
```
Comment (rianfowler) [inline src/applications/proxy-rewrite/proxy-rewrite-entry.jsx]: current modal attaches listeners to buttons in the header and footer to activate the modal; in the spirit of expediency we're reusing this functionality on the new footer component

File: src/platform/site-wide/sass/merger.scss [hunk 2]
```diff
@@ -14,6 +14,14 @@
   }
   .va-modal {
     z-index: 301;
+    font-size: 100%;
```
Comment (rianfowler) [inline src/platform/site-wide/sass/merger.scss]: this is to override some of the font behavior imported by the team site.

File: src/platform/site-wide/sass/style-consolidated.scss [hunk 3]
```diff
@@ -22,7 +21,7 @@ $image-path: "~uswds/src/img";
 
   // ----- VA MODULES (BUTTONS/CARDS/TABLES) ---- //
   @import "~@department-of-veterans-affairs/formation/sass/modules/m-alert";
-  @import "modules/m-crisis-line";
+  @import "modules/m-crisis-line-brand-consolidation";
```
Comment (rianfowler) [inline src/platform/site-wide/sass/style-consolidated.scss]: crisis line has a separate styling module on the WBC site.

File: src/platform/site-wide/va-footer/containers/Main.jsx [hunk 4]
```diff
@@ -26,8 +26,12 @@ export class Main extends React.Component {
     super(props);
     this.state = { isMobile: false };
     this.linkObj = groupBy(replaceDomainsInData(links), 'column');
+    this.state = {
+      isMobile: !isWideScreen(),
+    };
   }
-  componentWillMount() {
+  componentDidMount() {
```
Comment (rianfowler) [inline src/platform/site-wide/va-footer/containers/Main.jsx]: `componentWillMount` is deprecated in React 16. refactoring this now

File: src/platform/site-wide/va-footer/containers/Main.jsx [hunk 5]
```diff
@@ -37,10 +41,7 @@ export class Main extends React.Component {
       },
       false,
     );
-
-    this.setState({
```
Comment (rianfowler) [inline src/platform/site-wide/va-footer/containers/Main.jsx]: can't set state in `componentDidMount`- moving this into the constructor didn't impact the behavior.

File: src/platform/site-wide/va-footer/containers/Main.jsx [hunk 6]
```diff
@@ -278,6 +279,86 @@ export class Main extends React.Component {
             </ul>
           </div>
         </div>
+        <div
```
Comment (rianfowler) [inline src/platform/site-wide/va-footer/containers/Main.jsx]: crisis line modal was not ported into the new react footer that's used on team sites- this markup was taken verbatim from our existing html template.

File: src/platform/site-wide/va-footer/containers/Main.jsx [hunk 7]
```diff
@@ -37,10 +41,7 @@ export class Main extends React.Component {
       },
       false,
     );
-
-    this.setState({
-      isMobile: !isWideScreen(),
-    });
+    this.props.handleFooterDidMount();
```
Comment (rianfowler) [inline src/platform/site-wide/va-footer/containers/Main.jsx]: call mount handler here. assumes the header is loaded before the header.

**Comment by rianfowler** [comment]
Marking this WIP for now- this approach doesn't work on https://dev.va.gov/health/ which has an old `#modal-crisisline` that is conflicting with our new one.

**Review by bshyong — APPROVED** [review]
WIP work up to this point LGTM!

File: src/applications/proxy-rewrite/proxy-rewrite-entry.jsx [hunk 8]
```diff
@@ -76,6 +78,10 @@ document.addEventListener('DOMContentLoaded', _e => {
 // Mount react components
 const commonStore = createCommonStore();
 document.addEventListener('DOMContentLoaded', _e => {
+  const crisisModal = document.getElementById('modal-crisisline');
```
Comment (rianfowler) [inline src/applications/proxy-rewrite/proxy-rewrite-entry.jsx]: an ugly hack but faster than refactoring the modal id we're using across 2 sites. when we refactor this, will push to only use classNames.

**Comment by rianfowler** [comment]
@cehsu @bshyong @erikphansen This is ready for review. The only thing I added after your approval ben was deleting the existing #modal-crisisline if it exists.

**Pull request merged**